services/provider/server/server.go

```diff
@@ -487,8 +493,16 @@ func decodeAndValidateTicket(ticket string, pubKey *rsa.PublicKey) (*services.On
 		return nil, fmt.Errorf("failed to unmarshal onboarding ticket message. %v", err)
 	}
 
-	if ticketData.StorageQuotaInGiB > math.MaxInt {
-		return nil, fmt.Errorf("invalid value sent in onboarding ticket, storage quota should be greater than 0 and less than %v: %v", math.MaxInt, ticketData.StorageQuotaInGiB)
+	switch ticketData.SubjectRole {
+	case services.ClientRole:
+		if ticketData.StorageQuotaInGiB != nil {
+			quota := *ticketData.StorageQuotaInGiB
+			if quota > math.MaxInt {
+				return nil, fmt.Errorf("invalid value sent in onboarding ticket, storage quota should be greater than 0 and less than %v: %v", math.MaxInt, quota)
+			}
+		}
+	default:
+		return nil, fmt.Errorf("invalid onboarding ticket subject role")
```

Comment:
You are missing an (empty) case for the peer role. Without it, a ticket with a peer role will just go to the default case and will return an error

Reply:
I will be adding the case related to the peer role while implementing the rpc calls that we have for StorageClusterPeer, I don't think we should add it beforehand

Reply:
This pr intreduces the ability to generate peer tokens which means peer tokens should be decodable (as they are already encodable). Let's have the additional case to support it here